File: backend/ecommerce/src/main/java/com/example/ecommerce/dto/MomoResponse.java
package com.example.ecommerce.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MomoResponse {

    private String payUrl;

    private String qrCodeUrl;

    private String deeplink;

    private String orderId;

    private String requestId;

    private Integer resultCode;

    private String message;
}

File: ecommerce_app/lib/models/momo_response.dart
class MomoResponse {
  final String payUrl;
  final String? deeplink;
  final String? qrCodeUrl;
  final String orderId;
  final String requestId;
  final int resultCode;
  final String message;

  MomoResponse({
    required this.payUrl,
    this.deeplink,
    this.qrCodeUrl,
    required this.orderId,
    required this.requestId,
    required this.resultCode,
    required this.message,
  });

  factory MomoResponse.fromJson(Map<String, dynamic> json) {
    return MomoResponse(
      payUrl: json['payUrl'],
      deeplink: json['deeplink'],
      qrCodeUrl: json['qrCodeUrl'],
      orderId: json['orderId'],
      requestId: json['requestId'],
      resultCode: json['resultCode'],
      message: json['message'],
    );
  }
}
